sklearn/model_selection/_validation.py

```diff
@@ -1559,5 +1559,9 @@ def _aggregate_score_dicts(scores):
     {'a': array([1, 2, 3, 10]),
      'b': array([10, 2, 3, 10])}
     """
-    return {key: np.asarray([score[key] for score in scores])
-            for key in scores[0]}
+    return {
+        key: np.asarray([score[key] for score in scores])
+        if key.endswith(("time", "score"))
+        else [score[key] for score in scores]
```

Comment:
I would prefer to dispatch based on the type:

```py
    return {
        key: np.asarray([score[key] for score in scores])
        if isinstance(scores[0][key], numbers.Number)
        else [score[key] for score in scores]
        for key in scores[0]
    }
```

and then update the docstring:

> Aggregate a list of dicts to a dict of ndarray or list.

WTYD @glemaitre ?

Reply:
I made the change and merged so that PRs are not failing. We can have a followup PR if we want to go back to checking the key.

Reply:
Hmm... the only recent PR that was blocked was https://github.com/scikit-learn/scikit-learn/pull/17743